src/main/scala/rise/core/types/Kinds.scala

```diff
@@ -1,39 +1,77 @@
 package rise.core.types
 
-sealed trait Kind[+T, +I <: Kind.Identifier] {
+sealed trait Kind[+T, +I, KI <: Kind.Identifier] {
   def name :String
 }
 
+
 object Kind {
-  trait Identifier {
-    def name: String
+  trait Identifier { def name: String }
```

Comment:
In Scala 3 maybe this could be written using Match Types: https://dotty.epfl.ch/docs/reference/new-types/match-types.html

Reply:
Probably, yes!